internal/pkg/levee/propagation/propagation.go

```diff
@@ -52,8 +52,8 @@ func Dfs(n ssa.Node, conf *config.Config, taggedFields fieldtags.ResultType) Pro
 	}
 	maxInstrReached := map[*ssa.BasicBlock]int{}
 
-	record.visitReferrers(n, maxInstrReached, nil)
 	record.dfs(n, maxInstrReached, nil, false)
+	record.visitReferrers(n, maxInstrReached, nil)
```

Comment:
This was added in #214 to replace a check in `visit`, which was: 
```go
if n == prop.root {
    prop.visitReferrers(n, ...)
}
```
The intent of that check was just to ensure that the referrers of a `Source` node are visited.

A side effect of putting `visitReferrers` first is that the traversal order, as recorded by `preOrder` and `marked`, is no longer correct. Indeed, the root is marked *after* its referrers have been visited.

Having the correct order is necessary for the changes in this PR.